### PR TITLE
PF-543: prevent unintentional / hidden data from being saved to application

### PR DESCRIPTION
**Pull request by paulogiacomelli**
@dtorres-sf might need some more testing from your end on edge cases.

### CODE REVIEW

File: src/formGroup.vue [hunk 1]
```diff
@@ -32,6 +32,12 @@ export default {
 	name: "form-group",
 	components: fieldComponents,
 	mixins: [formMixin],
+	beforeDestroy() {
+		// if deleteDataOnHide is true, we remove the field and data from the model once the field is hidden.
+		if (this.options.deleteDataOnHide) {
```
Comment (dtorres-sf) [inline src/formGroup.vue]: This could be pretty risky. Does a wizard form `destroy` the pages that aren't visible? If so, this would get called every time a page is changed. Same with when we show success messages potentially or anytime we take down the form.

I think it is safer to actually do this sort of thing when visiblity changes based on when the function is evaluated. 

Additionally we also need to be able to reset this to the initial value when visibility is no longer false. So on show, we want to set the initial value, otherwise we could lose the initial value by enabling and then disabling the field

**Review by dtorres-sf — CHANGES_REQUESTED** [review]

File: src/formGenerator.vue [hunk 2]
```diff
@@ -124,11 +123,31 @@ export default {
...
+			// if the schema formOptions includes a deleteDataOnHide attribute:
+			// 1. we check if visibility became false on the field. If so, delete the model from that field
+			// 2. if became visible and we have an initial, we can set that initial back if the field not in model
+			if (this.options.deleteDataOnHide) {
+				if (!visible) {
+					this.$delete(this.model, field.model);
+				}
+				else if (visible && field.initial) {
+					if (!(field.model in this.model)) {
+						this.$set(this.model, field.model, field.initial);
```
Comment (dtorres-sf) [inline src/formGenerator.vue]: vue form generator allows for dotted paths to create complex option in model names, so `complex.model.name` will create nested objects. This is done via `setModelValueByPath` but it is essentially following the logic of `set` in lodash.

We have `vueSet` in our repo that maybe we want to add to the `utils` file in vfg.

File: src/formGenerator.vue [hunk 3]
```diff
@@ -124,11 +123,31 @@ export default {
...
+				visible = true;
+			}
 
-			if (isNil(field.visible)) return true;
+			// if the schema formOptions includes a deleteDataOnHide attribute:
+			// 1. we check if visibility became false on the field. If so, delete the model from that field
+			// 2. if became visible and we have an initial, we can set that initial back if the field not in model
+			if (this.options.deleteDataOnHide) {
+				if (!visible) {
+					this.$delete(this.model, field.model);
```
Comment (dtorres-sf) [inline src/formGenerator.vue]: This is actually the same problem as below, this will not work with a complex model name.

**Review by dtorres-sf — CHANGES_REQUESTED** [review]

**Comment by paulogiacomelli** [comment]
@dtorres-sf this is ready for re-review.

**Comment by paulogiacomelli** [comment]
new branch created.. less messy
